### PR TITLE
feature: Changes to add V8 and V9 themes together to support v8 and v…

### PR DESCRIPTION
**Pull request by v-rakeshsh**
#### Details

This feature is to bind V8 theme for existing Fluent ui components and V9 for new Fluent Ui components (next feature).

- So as part of Fluent UI V9 migration, we will not migrate entire components from V8 to V9, as we are following migration in incremental approach as V9 is also in development from fluent team.
- So to accommodate V8 and V9 components UI designs, we created V8 and V9 theme wrapped in root. So that V8 components will pickup v8 theme and V9 components (NEXT FEATURE) will pickup V9 theme.
- This feature also covers, high contrast themes for both v8 and v9 themes.
- Updated to latest V9 Fluent version @fluentui/react-components to 9.x.x
- Removed loadTheme from V8, as we are using theme as Wrapper instead of calling function.
- Created custom V9 Dark theme to accommodate high contrast option.

**Updated below test cases :**
src/tests/end-to-end/tests/details-view/headings.test.ts
src/tests/end-to-end/tests/details-view/reflow-ui.test.ts

During e2e test cases encountered below errors for above 2 files:
GetImage

So from investigation, we can observe its false positive for high contrast, and there is existing issues reported for transparent background. 
https://github.com/dequelabs/axe-core/issues/3464
And the issue we are seeing is with **Assessment link** in high contrast mode, inspecting this link in high contrast mode, we can see background is in transparent.

- Removed loadTheme test case, as we are not using loadTheme anymore.

##### Motivation

To keep AI Web to accommodate both V8 and V9 components with respective theme.

#### Pull request checklist

- [ ] Addresses an existing issue: #0000
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [x] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/tests/unit/tests/common/components/theme.test.tsx [hunk 1]
```diff
@@ -43,39 +45,29 @@ describe('ThemeInner', () => {
...
 
     test.each(testStub)(
         'componentDidMount: is high contrast mode enabled: %s',
         (enableHighContrast: boolean) => {
-            const theme = enableHighContrast ? HighContrastTheme : DefaultTheme;
             const userConfigurationStoreData = { enableHighContrast } as UserConfigurationStoreData;
             render(<ThemeInner {...props} storeState={{ userConfigurationStoreData }} />);
-
-            expect(loadThemeMock).toHaveBeenCalledWith(theme);
+            const hasThemeProp = getMockComponentClassPropsForCall(ThemeProvider).theme;
```
Comment (madalynrose) [inline src/tests/unit/tests/common/components/theme.test.tsx:57]: should we be checking for both the v8 theme and the v9 theme here?

File: src/tests/unit/tests/common/components/theme.test.tsx [hunk 2]
```diff
@@ -43,39 +45,29 @@ describe('ThemeInner', () => {
...
         'componentDidUpdate: is high contrast mode enabled: %s',
         (enableHighContrast: boolean) => {
-            const theme = enableHighContrast ? HighContrastTheme : DefaultTheme;
             const { rerender } = render(<ThemeInner {...props} />);
 
-            loadThemeMock.mockReset();
             const userConfigurationStoreData = { enableHighContrast } as UserConfigurationStoreData;
             rerender(<ThemeInner {...props} storeState={{ userConfigurationStoreData }} />);
-            expect(loadThemeMock).toHaveBeenCalledWith(theme);
+            const hasThemeProp = getMockComponentClassPropsForCall(ThemeProvider).theme;
```
Comment (madalynrose) [inline src/tests/unit/tests/common/components/theme.test.tsx:69]: should we be checking for both the v8 theme and the v9 theme here?

**Review by madalynrose — CHANGES_REQUESTED** [review]
a couple small adjustments but this looks solid overall!

File: src/tests/end-to-end/tests/details-view/headings.test.ts [hunk 3]
```diff
@@ -79,7 +79,7 @@ describe('Details View -> Assessment -> Headings', () => {
                     headingsPage,
                     detailsViewSelectors.mainContent,
                 );
-                expect(results).toStrictEqual([]);
+                expect(results).toMatchSnapshot();
```
[the remark this answers is not in the document]
Comment (v-rakeshsh) [reply src/tests/end-to-end/tests/details-view/headings.test.ts:87]: Hi @madalynrose  added comments, please re-review. Thanks

File: src/tests/end-to-end/tests/details-view/reflow-ui.test.ts [hunk 4]
```diff
@@ -233,7 +233,7 @@ describe('Details View ->', () => {
         await detailsViewPage.waitForHighContrastMode(highContrastMode);
 
         const results = await scanForAccessibilityIssues(detailsViewPage, '*');
-        expect(results).toHaveLength(0);
+        expect(results).toMatchSnapshot();
```
[the remark this answers is not in the document]
Comment (v-rakeshsh) [reply src/tests/end-to-end/tests/details-view/reflow-ui.test.ts:241]: Hi @madalynrose  added comments, please re-review. Thanks

File: src/tests/unit/tests/common/components/theme.test.tsx (hunk 1 above)
Comment (v-rakeshsh) [reply src/tests/unit/tests/common/components/theme.test.tsx]: Hi @madalynrose agree, added check for v9 as well now, please re-review. Thanks

File: src/tests/unit/tests/common/components/theme.test.tsx (hunk 2 above)
Comment (v-rakeshsh) [reply src/tests/unit/tests/common/components/theme.test.tsx]: Hi @madalynrose agree, added check for v9 as well now, please re-review. Thanks

**Review by madalynrose — CHANGES_REQUESTED** [review]
one small nit but this looks great!

File: src/tests/end-to-end/tests/details-view/headings.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (v-rakeshsh) [reply src/tests/end-to-end/tests/details-view/headings.test.ts:88]: Hi @madalynrose added in comment now, please re-review, also added in PR description as well.

File: src/tests/end-to-end/tests/details-view/reflow-ui.test.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (v-rakeshsh) [reply src/tests/end-to-end/tests/details-view/reflow-ui.test.ts:242]: Hi @madalynrose added in comment now, please re-review, also added in PR description as well.

**Review by madalynrose — APPROVED** [review]

**Pull request merged**